### PR TITLE
fix(secrets): change entropy limit in Combinator plugin

### PR DESCRIPTION
**Pull request by marynaKK**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Sets a different limit for entropy for the standalone plugin vs. the combinator plugin.

The change would cause finding fewer false positive secrets in source code, but leaves the Iac findings as it.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [ ] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: tests/secrets/test_plugin.py [hunk 1]
```diff
@@ -24,7 +23,7 @@ def test_negative_entropy_value(self):
         self.assertEqual(0, len(result))
 
     def test_popular_kubernetes_manifest_password(self):
-        result = self.plugin.analyze_line("mock.yaml", 'pwd: "correcthorsebatterystaple"', 5)
+        result = self.plugin.analyze_line("mock.yaml", 'pwd: "correcthorsebatterystaple"', 5)  # fails now
```
Comment (marynaKK) [inline tests/secrets/test_plugin.py:26]: this one will fail with the current limit

File: checkov/secrets/runner.py [hunk 2]
```diff
@@ -60,7 +60,7 @@
 }
 CHECK_ID_TO_SECRET_TYPE = {v: k for k, v in SECRET_TYPE_TO_ID.items()}
 
-ENTROPY_KEYWORD_LIMIT = 3
+ENTROPY_KEYWORD_LIMIT = 4.5
```
[the remark this answers is not in the document]
Comment (marynaKK) [reply checkov/secrets/runner.py]: moved it 😄  thanks!

**Review by gruebel — APPROVED** [review]
nice, looks good 🍹

File: checkov/secrets/runner.py [hunk 3]
```diff
@@ -69,62 +68,36 @@
...
+                {'name': 'IbmCloudIamDetector'},
+                {'name': 'MailchimpDetector'},
+                {'name': 'PrivateKeyDetector'},
+                {'name': 'SlackDetector'},
+                {'name': 'SoftlayerDetector'},
+                {'name': 'SquareOAuthDetector'},
+                {'name': 'StripeDetector'},
+                {'name': 'TwilioKeyDetector'},
+                {'name': 'Base64HighEntropyString'},  # default limit from secrets lib is 4.5
+                {'name': 'HexHighEntropyString'},  # default limit from secrets lib is 4.5
```
Comment (nimrodkor) [inline checkov/secrets/runner.py:97]: @marynaKK What filters these plugins out for IaC files?

Comment (nimrodkor) [reply checkov/secrets/runner.py:97]: It seems that's not tested - can you add a UT that scans a directory with 1 IaC file and 1 source file, and ensure the custom plugin runs on the IaC and not the source file, and the Entropy plugins run on the source file and not on the IaC?

Comment (marynaKK) [reply checkov/secrets/runner.py]: I didn't add a directory - I've tried to preserve the current testing folders. But I did name the UTs which are checking for no FP.

**Comment by marynaKK** [comment]
@nimrodkor does it look OK now? 
thanks! 🌸

File: checkov/secrets/plugins/entropy_keyword_combinator.py [hunk 4]
```diff
@@ -34,18 +38,16 @@ def analyze_line(
         one of the entropy scanners find a match (on a line which was already matched by keyword plugin) - it is returned.
         for source code files run and merge the two plugins.
         """
-        entropy_matches = set()
+        is_iac = True if f".{filename.split('.')[-1]}" not in SOURCE_CODE_EXTENSION else False
```
Comment (nimrodkor) [inline checkov/secrets/plugins/entropy_keyword_combinator.py:41]:
```suggestion
        is_iac = f".{filename.split('.')[-1]}" not in SOURCE_CODE_EXTENSION
```

**Review by nimrodkor — APPROVED** [review]
Nice!

File: checkov/secrets/plugins/entropy_keyword_combinator.py (hunk 4 above)
Comment (marynaKK) [reply checkov/secrets/plugins/entropy_keyword_combinator.py]: done! :) ty

**Pull request merged**